src/js/mqtt.js

```diff
@@ -191,37 +260,38 @@ MqttClient.prototype.subscribe = function(topic, options, callback) {
  * @param {Function} callback
  */
 MqttClient.prototype.unsubscribe = function(topic, callback) {
-  var buf = this._handle._getUnsubscribe(topic, {
-    id: this._msgId++,
-  });
-  this._write(buf, callback);
-};
-
-/**
- * @method unsubscribe
- * @param {Boolean} isForce
- * @param {Function} callback
- */
-MqttClient.prototype.end = function(isForce, callback) {
-  clearTimeout(this._ttl);
-
-  var buf = this._handle._getDisconnect();
-  if (isForce) {
-    this._write(buf);
-    this._socket.end();
-  } else {
-    this._write(buf, () => {
-      this._socket.end();
+  callback = callback || noop;
+  if (!Array.isArray(topic)) {
+    topic = [topic];
+  }
+  try {
+    var buf = this._handle._getUnsubscribe(topic, {
+      id: this._msgId++,
     });
+    this._write(buf, callback);
+  } catch (err) {
+    callback(err);
   }
```

Comment:
Why use `try-catch` here?

Reply:
this._handle._getUnsubscribe may throw an error which means buffer is not enough